component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -1658,4 +1669,114 @@ private long getUnlockTimeInMilliSeconds(AuthenticatedUser authenticatedUser) th
                     username, e);
         }
     }
+
+    private void publishPostSMSOTPGeneratedEvent(HttpServletRequest request, AuthenticationContext context)
+            throws AuthenticationFailedException {
+
+        Map<String, Object> eventProperties = new HashMap<>();
+        eventProperties.put(IdentityEventConstants.EventProperty.CORRELATION_ID, context.getCallerSessionKey());
+        AuthenticatedUser authenticatedUser = (AuthenticatedUser) context.getProperty(SMSOTPConstants
+                .AUTHENTICATED_USER);
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_NAME, authenticatedUser.getUserName());
+        eventProperties.put(IdentityEventConstants.EventProperty.TENANT_DOMAIN, context.getTenantDomain());
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_STORE_DOMAIN, authenticatedUser
+                .getUserStoreDomain());
+        eventProperties.put(IdentityEventConstants.EventProperty.APPLICATION_NAME, context.getServiceProviderName());
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_AGENT, request.getHeader(
+                SMSOTPConstants.USER_AGENT));
+        if (request.getParameter(SMSOTPConstants.RESEND) != null) {
+            eventProperties.put(IdentityEventConstants.EventProperty.RESEND_CODE,
+                    request.getParameter(SMSOTPConstants.RESEND));
+        } else {
+            eventProperties.put(IdentityEventConstants.EventProperty.RESEND_CODE, false);
+        }
+
+        eventProperties.put(IdentityEventConstants.EventProperty.GENERATED_OTP, context.getProperty(
+                SMSOTPConstants.OTP_TOKEN));
+
+        long otpGeneratedTime = (long) context.getProperty(SMSOTPConstants.SENT_OTP_TOKEN_TIME);
+        eventProperties.put(IdentityEventConstants.EventProperty.OTP_GENERATED_TIME, otpGeneratedTime);
+
+        String otpExpiryDuration = SMSOTPUtils.getTokenExpiryTime(context);
+        if (otpExpiryDuration != null) {
+            long expiryTime = otpGeneratedTime + Long.parseLong(otpExpiryDuration);
+            eventProperties.put(IdentityEventConstants.EventProperty.OTP_EXPIRY_TIME, expiryTime);
+        }
+
+        if (request.getHeader(SMSOTPConstants.X_FORWARDED_FOR) != null) {
+            eventProperties.put(IdentityEventConstants.EventProperty.CLIENT_IP, request.getHeader(
+                    SMSOTPConstants.X_FORWARDED_FOR).split(",")[0]);
+        } else {
+            eventProperties.put(IdentityEventConstants.EventProperty.CLIENT_IP, request.getRemoteAddr());
+        }
+
+        Event postOtpGenEvent = new Event(IdentityEventConstants.Event.POST_GENERATE_SMS_OTP, eventProperties);
+        try {
+            SMSOTPServiceDataHolder.getInstance().getIdentityEventService().handleEvent(postOtpGenEvent);
+        } catch (IdentityEventException e) {
+            String errorMsg = "An error occurred while triggering the event. " + e.getMessage();
```

Comment:
Shall we add a more descriptive error message, whether this post-event in SMSOTP generation flow something like this?

Reply:
Fixed in 7d65437416ff6a2f9c995097fb0958fc95a58cf3